### PR TITLE
fix: for "Nonce Has Already Been Used" Issue During Deployment on Calibrationnet

### PR DESCRIPTION
**Pull request by karlem**
#### Issue
- **Problem:** The error arises because the nonce, which uniquely identifies each transaction from the sender, is reused while deploying multiple contracts on the calibrationnet. This issue does not occur on a local node, likely because transactions are processed immediately in a single-node environment.
- **Cause:** On a public testnet like calibrationnet, network latency, transaction propagation time, and node synchronization can lead to situations where a new transaction is sent before the previous one is fully confirmed, resulting in nonce reuse.
- **Local Node Success:** On a local node, deployments succeed because transactions are mined instantly, preventing any nonce conflicts.

#### Solution
The solution is to manage the nonce automatically by calling getTransactionCount with the pending argument to align with Filecoin.

### CODE REVIEW

**Comment by cryptoAtwill** [comment]
Interesting. Even though we are targeting the same node, ideally with one confirmation, it should have updated the nonce usage. Unless the node is load balanced?

**Comment by karlem** [comment]
> Unless the node is load balanced?

It is.


> If this is just about confirmation, then increase to 3 might slow down the deployment 3 times. Or maybe we can try a `try-catch` and handle the error to wait just 1 block time.

Sure we can try. Or we can increase the confirmation to 2. This is just a hot fix.

**Review by cryptoAtwill — APPROVED** [review]
@karlem I tried it, and it's working.

But it's really slow deploying all the contracts. We have 27 contracts (maybe more to come?), then 27 * 3 * (30 seconds) ~= 40 min. I would say speed is important too.

**Review by cryptoAtwill — COMMENTED** [review]
I approved it by accident, my bad. But I think the above comment might be helpful.

**Comment by karlem** [comment]
@cryptoAtwill I think using 2 confirmations with a try-catch mechanism is the sweet spot. I tried 1 confirmation, but it resulted in many retries and also took a while. What do you think?

**Comment by karlem** [comment]
Changed to manually managing nonce.

File: .github/workflows/contracts-deployment-test.yaml
Comment (raulk) [inline .github/workflows/contracts-deployment-test.yaml:1]: Revert formatting changes?

**Review by raulk — APPROVED** [review]

**Pull request merged**